scripts/kconfig/kconfigfunctions.py

```diff
@@ -0,0 +1,75 @@
+#
+# Copyright (c) 2018-2019 Linaro
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+
+# Types we support
+# 'string', 'int', 'hex', 'bool'
+conf_file = os.path.join(os.environ['PROJECT_BINARY_DIR'], "include/generated/generated_dts_board.conf")
+
+dt_defines = {}
+if os.path.isfile(conf_file):
+    for line in open(conf_file, 'r', encoding='utf-8'):
```

Comment:
'with' must be used.

AFAICT this code has a resource leak[0].

[0] https://stackoverflow.com/a/25070939

Reply:
Changed to use 'with'